modelexpress_client/python/modelexpress/refit/reshard/rendezvous.py

```
@@ -176,14 +177,25 @@ def build_sources(tensors: list) -> tuple:
     sources = {}
     session_to_agent = {}
     session_to_device = {}
     for t in tensors:
         dtype = _torch_dtype(t.dtype)
-        shards = []
+        by_box: dict = {}
         for s in t.shards:
+            by_box.setdefault((tuple(s.shard_offset), tuple(s.shape)), []).append(s)
+        shards = []
+        for group in by_box.values():
+            s = random.choice(group)
```

Ruff (0.16.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)